src/sync/channel.zig

```diff
@@ -37,18 +39,34 @@ pub fn Channel(comptime T: type) type {
             if (self.closed.load(.Monotonic)) {
                 return error.ChannelClosed;
             }
-            var buffer = self.buffer.lock();
-            defer buffer.unlock();
-            try buffer.mut().append(value);
-            self.hasValue.signal();
+            var buffer_lock = self.buffer.lock();
+            defer buffer_lock.unlock();
+
+            var buffer: *std.ArrayList(T) = buffer_lock.mut();
```

Comment:
readability + you dont get autocomplete for just .mut() since it doesnt figure out the type - in this case its not a big deal since its just `append()`, but in other situations the autocomplete is really nice to have - so we just keep it consistent here